src/worker/network.rs

```diff
@@ -348,23 +349,24 @@ impl NetworkWorker {
     }
 
     /// Map socket index to a local port and address. The address octets are
-    /// incremented every 100 sockets, whithin this interval the local port
+    /// incremented every ports_per_addr sockets, whithin this interval the local port
     /// is incremented.
     fn get_local_addr_port(
         &self,
         addr: Ipv4Address,
-        index: usize,
+        ports_per_addr: u16,
+        index: u64,
     ) -> (IpAddress, u16) {
-        // 254 (a2 octet) * 254 (a3 octet) * 100 (port)
+        // 254 (a2 octet) * 254 (a3 octet) * ports_per_address (port)
         // gives us maximum 6451600 connections that could be opened
-        let local_port = 49152 + (index % 100) as u16;
+        let local_port = 49152 + (index % ports_per_addr as u64) as u16;
```

Comment:
I set `ports_per_addr` to `u16`, but I am always converting it to `u64`. Seems a bit strange.

Reply:
it is a little strange, but upcasting at least means we can guarantee that we're not losing precision. It's probably fine as-is.